frontend/packages/shared/src/components/InputActionWrapper/InputActionWrapper.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from 'react';
+import { CheckmarkIcon, TrashIcon, PencilWritingIcon } from '@altinn/icons';
+import { useText } from '../../../../ux-editor/src/hooks/index';
+import { Button, ButtonProps } from '@digdir/design-system-react';
+import classes from './InputActionWrapper.module.css';
+
+type AvailableAction = 'edit' | 'save' | 'delete';
+export type ActionGroup = 'editMode' | 'hoverMode' | 'standBy';
+
+const actionGroupMap: Record<ActionGroup, AvailableAction[]> = {
+  editMode: ['save', 'delete'],
+  hoverMode: ['edit', 'delete'],
+  standBy: [],
+};
+
+const actionToIconMap: Record<AvailableAction, React.ReactNode> = {
+  edit: <PencilWritingIcon />,
+  save: <CheckmarkIcon />,
+  delete: <TrashIcon />,
+};
+
+export type InputActionWrapperProps = {
+  children: React.ReactElement;
+  mode?: ActionGroup;
+  onEditClick: () => void;
+  onDeleteClick: () => void;
+  onSaveClick: () => void;
+};
+
+export const InputActionWrapper = ({
+  mode,
+  children,
+  onEditClick,
+  onDeleteClick,
+  onSaveClick,
+  ...rest
+}: InputActionWrapperProps): JSX.Element => {
+  const t = useText();
+  const [actions, setActions] = useState<AvailableAction[]>(actionGroupMap[mode || 'standBy']);
+
+  const handleFocus = (): void => {
+    setActions(actionGroupMap['editMode']);
+  };
+
+  const handleBlur = (): void => {
+    setActions(actionGroupMap['standBy']);
+  };
+
+  const handleMouseLeave = (): void => {
+    const isEditMode = actions.includes('save');
+    if (isEditMode) return;
+    handleBlur();
+  };
+
+  const handleActionClick = (action: AvailableAction): void => {
+    switch (action) {
+      case 'delete':
+        onDeleteClick();
+        break;
+      case 'save':
+        onSaveClick();
+        handleBlur();
+        break;
+      case 'edit':
+        onEditClick();
+        setActions(actionGroupMap['editMode']);
+      default:
+        break;
+    }
+  };
+
+  const handleHover = (): void => {
+    const isInStandByMode = mode === 'standBy';
+    if (isInStandByMode) {
+      setActions(actionGroupMap['hoverMode']);
+    }
+  };
+
+  const actionToAriaLabelMap: Record<AvailableAction, string> = {
+    edit: t('general.edit'),
+    delete: t('general.delete'),
+    save: t('general.save'),
+  };
+
+  const actionToColorMap: Record<AvailableAction, ButtonProps['color']> = {
+    edit: 'first',
+    save: 'success',
+    delete: 'danger',
+  };
+
+  const handleSize = (action: AvailableAction): 'small' | 'medium' | 'large' => {
+    return 'large';
+  };
+
+  return (
+    <div className={classes.buttons} onMouseOver={handleHover} onMouseLeave={handleMouseLeave}>
+      {React.cloneElement(children, {
+        ...rest,
+        onFocus: handleFocus,
+      })}
+      {actions.map((action) => (
+        <Button
+          variant='quiet'
+          size={handleSize(action)}
```

Comment:
Ref. previous comment, I think we should hard code the value directly here since it is always large. Then we avoid one method execution. 😊 
```suggestion
          size="large"
```